### ISSUE TITLE
Expansion version in save file

### ISSUE THREAD

**Issue by Ninjdai1** [rh-hideout/pokeemerald-expansion#3829]
### Description

TL;DR: Similar to #3454 in the save file

The goal of this would be to make the expansion's version accessible to programs similar to pkhex:
I'm currently in the making of an expansion-compatible savefile editor. Problem is, some mons and items are not available depending on the version, which would probably cause issues if a mon, moves or items were added in a version they weren't in. Also, if #3438 or similar were to get merged, it would make that kind of software able to change the offsets it uses to get the data.

I know, as it would most likely change a saveblock, this will probabl get rejected, but I still wanted to bring the idea.
Thanks for reading :3

### Discord contact info

ninjdai

**Comment by mrgriffin** [comment]
You probably want your save editor to open both the ROM and the save, and that way you could read the Expansion version out of the [RHH ROM header](https://github.com/rh-hideout/pokeemerald-expansion/blob/master/src/rom_header_rhh.c). The [GF ROM header](https://github.com/rh-hideout/pokeemerald-expansion/blob/master/src/rom_header_gf.c) will give you pointers to things like `gMoveNames` and `gSpeciesInfo` which can help you identify which moves/species/etc are available.

Of course if a hack has added additional stuff to `gSpeciesInfo` or whatever, then the Expansion version isn't enough to tell you how to interpret the data, and atm there's nothing in the ROM or save which would.

**Comment by mrgriffin** [comment]
Oh, you might need things like `MOVES_COUNT` added to the RHH ROM header so that you don't have to resort to hacky things with `gMoveNames` to work out how many moves there are—feel free to raise a PR with that :)
